### ISSUE TITLE
Egon.io as Visual Studio Code plugin

### ISSUE THREAD

**Issue by phpmaple** [WPS/domain-story-modeler#130]
It would be great if there is a plugin in vscode

**Comment by hschwentner** [comment]
I agree; would love to have this myself. Currently, we don’t have the time, but hopefully after the port to Angular is finished and v2 is out of the door.

**Comment by phpmaple** [comment]
looking forward to it

**Comment by hofstef** [comment]
We're probably going to have a look at this at the beginning of 2025.

**Comment by peterhnm** [comment]
I took a quick look at this and think I can provide some support here. I have already done a lot with VS Code plugins and bpmn.io e.g.: https://github.com/Miragon/miranum-ide  

@hofstef do you have time to talk about the requirements?

![image](https://github.com/user-attachments/assets/7ff17168-4348-42e1-82ef-da7b909c2bb5)

**Comment by hofstef** [comment]
I've opened disccusion #189 where we can collect requirements, constraints, and solution ideas for the VS Code extension. This will probably lead to several new issues. Hence, I thought it is more handy to use the discussion feature as an umbrella.

**Comment by dominikhorn93** [comment]
We’ve done some work on our side, and @peterhnm has created an initial beta draft after performing some general repository restructuring to create a monorepo structure. You can download the first beta draft of the plugin [here](https://github.com/Miragon/egon.io/actions/runs/14172338012/artifacts/2851163740) and take a look at the fork [here](https://github.com/Miragon/egon.io/tree/feat/add-vscode-plugin). Still some features missing like replaying stories and reordering them or custom types.
